src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11213,11 +11227,6 @@ private bool ScanExplicitlyTypedLambda(Precedence precedence)
                             return foundParameterModifier;
 
                         case SyntaxKind.CommaToken:
-                            if (foundParameterModifier)
-                            {
-                                return true;
-                            }
```

Comment:
what happened here?

Reply:
This case was previously used to treat `(ref x,` as the start of an explicitly typed lambda. If we also allow those cases when there is an explicit return type, we'd treat `F(ref x,` as the start of a lambda (with return type `F`) rather than a call to `F()`. The simplest change seemed to be look ahead for `) =>` in all cases as we were doing for cases with parameters with no modifiers.

Reply:
as above, i'd put a comment to this effect in this method to help make it clearer for maintenance.